modules/apps/expando/expando-test/src/testIntegration/java/com/liferay/expando/service/test/ExpandoValueLocalServiceTest.java

```diff
@@ -187,6 +194,58 @@ public void testAddStringArrayValue() throws Exception {
 		Assert.assertEquals("two, three", data[1]);
 	}
 
+	@Test
+	public void testAddValuesWithFlushInBetween() throws Throwable {
+		ExpandoColumn column1 = ExpandoTestUtil.addColumn(
+			_expandoTable, "Test Column1", ExpandoColumnConstants.STRING);
+
+		ExpandoColumn column2 = ExpandoTestUtil.addColumn(
+			_expandoTable, "Test Column2", ExpandoColumnConstants.STRING);
+
+		long classPK = CounterLocalServiceUtil.increment();
+
+		// Create initial ExpandoValue for each column
+
+		ExpandoTestUtil.addValue(
+			_expandoTable, column1, classPK, "column1-one");
+		ExpandoTestUtil.addValue(
+			_expandoTable, column2, classPK, "column2-one");
+
+		// Remove cache to force  ExpandoValuePeristence find methods be
+		// executed against the database to force flush in between.
+
+		_entityCache.clearCache(ExpandoValueImpl.class);
+		_finderCache.clearCache(ExpandoValueImpl.class);
```

Comment:
@marianoalvarosaiz I see here you need to clean up the cache to make sure the later method invocation does the re-fetch which leads to a query that flushes the session. But is this happening in one of the reportings? Like when importing data?

Reply:
@dantewang 
It does happen with Andre Fabbro use case. His reproduction steps are:

    Create some User custom fields.
    Create a new Authenticator that updates the custom fields when logging in.
    If a try to log in I get no error, but if I restart the server or clean the cache I get an error when I log.

Reply:
But is the same with other LPP, if you wait enough time after creating the first expando values and you have two different expando columns whose value is going to be modified sequentially you get the same exception.